app/controllers/Checkout.scala

```diff
@@ -47,12 +48,13 @@ object Checkout extends Controller with LazyLogging {
         for (idUserOpt <- getIdentityUserByCookie(request))
           yield BadRequest(views.html.checkout.payment(formWithErrors, userIsSignedIn = idUserOpt.isDefined))
       },
-      userData => {
+      subscriptionData => {
         val idUserOpt = AuthenticationService.authenticatedUserFor(request)
-        CheckoutService.processSubscription(userData, idUserOpt).map {
+        CheckoutService.processSubscription(subscriptionData, idUserOpt).map {
           case CheckoutResult(_, guestUser: GuestUser, _) =>
             Some(FinishAccountForm().bind(guestUser.toFormParams))
-          case CheckoutResult(_, _:MinimalIdUser, _) =>
+          case CheckoutResult(_, registeredUser:MinimalIdUser, _) =>
+            updateStoredRegisteredUserDetails(subscriptionData.personalData, registeredUser.id, request.cookies.find(_.name == "SC_GU_U"))
```

Comment:
Why not moving this bit of logic into `CheckoutService.processSubscription`? You could pass authCookie along with the other two arguments. I think it would help making the controller code a bit more concise.